packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -1,18 +1,281 @@
+	// find param with the same id and update it
+	state.configParams = state.configParams?.map((p) => {
+		if (p.id === param.id) {
+			return param;
+		}
+		return p;
+	});
+	emit('update-state', state);
+};
+
+const updateConfigInitial = (initial) => {
+	const state = _.cloneDeep(props.node.state);
+	// find initial with the same target and update it
+	state.configInitials = state.configInitials?.map((i) => {
+		if (i.target === initial.target) {
+			return initial;
+		}
+		return i;
+	});
+	emit('update-state', state);
+};
+
+const saveConfiguration = async () => {
+	if (model.value) {
+		const newModel = _.cloneDeep(model.value);
+		if (newModel.semantics) {
+			newModel.semantics.ode.initials = configInitials.value;
+			newModel.semantics.ode.parameters = configParams.value;
+		}
+
+		const data = await createModelConfiguration(
+			model.value.id,
+			configName.value,
+			configDescription.value,
+			newModel
+		);
+
+		setTimeout(async () => {
+			const modelConfig = await getModelConfigurationById(data.id);
+			configCache.value[modelConfig.id] = modelConfig;
+
+			emit('append-output-port', {
+				type: ModelConfigOperation.outputs[0].type,
+				label: modelConfig.name,
+				value: modelConfig.id,
+				isSelected: false,
+				state: {
+					modelId: modelConfig.modelId,
+					configName: modelConfig.name,
+					configDescription: modelConfig.description,
+					configInitials: modelConfig.configuration.semantics?.ode.initials,
+					configParams: modelConfig.configuration.semantics?.ode.parameters
+				}
+			});
+		}, 1000);
+	}
+};
+
+const debouncedUpdateState = _.debounce(updateState, 500);
+
+watch(
+	() => props.node.state,
+	() => {
+		configInitials.value = props.node.state.configInitials;
+		configParams.value = props.node.state.configParams;
+	},
+	{ immediate: true }
+);
+
+const onUpdateOutput = (id) => {
+	emit('select-output', id);
+};
+
+const onUpdateSelection = (id) => {
+	const outputPort = _.cloneDeep(props.node.outputs?.find((port) => port.id === id));
+	if (!outputPort) return;
+	outputPort.isSelected = !outputPort?.isSelected;
+	emit('update-output-port', outputPort);
+};
+
+watch(
+	() => props.node.active,
+	() => {
+		// Update selected output
+		if (props.node.active) {
+			selectedOutputId.value = props.node.active;
+		}
+		// Fill the form with the config data
+		configName.value = props.node.state.configName;
+		configDescription.value = props.node.state.configDescription;
+		configInitials.value = props.node.state.configInitials;
+		configParams.value = props.node.state.configParams;
+	},
+	{ immediate: true }
+);
+
+const lazyLoadModelConfig = async (configId: string) => {
+	if (!configId || configCache.value[configId]) return;
+
+	const config = await getModelConfigurationById(configId);
+	if (config) {
+		configCache.value[configId] = config;
+	}
+};
+
+watch(
+	() => selectedConfigId.value,
+	() => {
+		lazyLoadModelConfig(selectedConfigId.value);
+	},
+	{ immediate: true }
+);
 </script>
```

Comment:
Can we do a bit or reorganization for the <script> section to make things more idiomatic?

We typically have the pattern of:
- imports
- props/emits declarations
- refs/computed declarations
- functions
- watchers
- lifecycle-hooks
- 
You have things a bit more intermingled.

Reply:
reorganized as suggested